### ISSUE TITLE
Add missing ack trigger in vpar support

### ISSUE THREAD

**Issue by cnvogelg** [FrodeSolheim/fs-uae#93]
First of all thanks for adding vpar support in the upstream tree!

Unfortunately, it currently lacks the possibility to trigger the ACK/INT2 irq. Without that feature it is not possible to emulate advanced parallel port hardware like the plipbox.

What is missing in current code are two things:

* vpar needs to be polled in regular intervals to find out if it has a pending ACK irq waiting. I must admit that I am not totally sure were to place such a call in the Amiga emulation as it should be polled in intervals that resemble the checking of the IRQ in the emulated CPU itself. As I was no able to find a suitable way to integrate it into the CPU core I used the CIA hsync call to poll the flag:

```
diff --git a/src/cia.cpp b/src/cia.cpp
index b0b53d0..c085da5 100644
--- a/src/cia.cpp
+++ b/src/cia.cpp
@@ -720,8 +720,11 @@ static void resetwarning_check (void)
        }
 }

+extern void vpar_update(void);
+
 void CIA_hsync_prehandler (void)
 {
+       vpar_update();
 }

 static void keyreq (void)
```

For a proper integration it might be useful to add a generic ```parallel_poll_ack()```call into UAE core and then map to ```vpar_update()``` inside ```od-fs/parser.cpp```.

* The second thing that needs to be changed is the auto-ack that is done whenever a byte is submitted to the parallel routines. In vpar a data write/read strobes but does not fake an vpar immediately. Ack is sent by the remote device.

In my prototype code I simply disabled all auto-ack'ing:

```
diff --git a/src/cia.cpp b/src/cia.cpp
index b0b53d0..c085da5 100644
--- a/src/cia.cpp
+++ b/src/cia.cpp
@@ -1329,7 +1332,7 @@ static void WriteCIAA (uae_u16 addr, uae_u8 val)
                        cia_parallelack ();
                } else if (isprinter() < 0) {
                        parallel_direct_write_data (val, ciaadrb);
-                       cia_parallelack ();
+            /* CV: cia_parallelack (); */
 #ifdef ARCADIA
                } else if (arcadia_bios) {
                        arcadia_parport (1, ciaaprb, ciaadrb);
```

For a proper integration a generic ```parallel_ack()```function can be introduced into UAE's core and then parser.cpp calls ```cia_parallelack()```for all modes except vpar.

**Comment by FrodeSolheim** [comment]
I have implemented these fixes in https://github.com/FrodeSolheim/fs-uae/commit/8033738336dd09da76e64dd83dae943f27dfb755 (in theory). If you can, please test, and let me know if you spot something fishy in the commit, or if something is missing!

**Comment by FrodeSolheim** [comment]
The mentioned fix is included in 2.7.15dev. Please let me know if it works as intended :)

**Comment by cnvogelg** [comment]
Yes, works like a charm!
Thanks for fixing.
